// Dstl (c) Crown Copyright 2024
//
// [START declaration]
syntax = "proto3";
package sapient_msg.bsi_flex_335_v2_0;

import "google/protobuf/timestamp.proto";

import "sapient_msg/bsi_flex_335_v2_0/alert.proto";
import "sapient_msg/bsi_flex_335_v2_0/alert_ack.proto";
import "sapient_msg/bsi_flex_335_v2_0/detection_report.proto";
import "sapient_msg/bsi_flex_335_v2_0/error.proto";
import "sapient_msg/proto_options.proto";
import "sapient_msg/bsi_flex_335_v2_0/registration.proto";
import "sapient_msg/bsi_flex_335_v2_0/registration_ack.proto";
import "sapient_msg/bsi_flex_335_v2_0/status_report.proto";
import "sapient_msg/bsi_flex_335_v2_0/task.proto";
import "sapient_msg/bsi_flex_335_v2_0/task_ack.proto";
// [END declaration]

// [START java_declaration]
option java_multiple_files = true;
option java_package = "uk.gov.dstl.sapientmsg.bsiflex335v2";
option java_outer_classname = "SapientMessageProto";
// [END java_declaration]


// [START options]
option (file_options) = {
  standard_version: "BSI_Flex_335_v2.0"
};
// [END options]


// [START messages]
message SapientMessage {
    optional google.protobuf.Timestamp timestamp = 1 [(field_options) = {is_mandatory: true}]; // UTC timestamp the message was sent
    optional string node_id = 2 [(field_options) = {is_mandatory: true, is_uuid: true}]; // UUID for the node which is sending the message
    optional string destination_id = 3 [(field_options) = {is_uuid: true}]; // UUID for the node being tasked
    oneof content {
        option (oneof_options) = {is_mandatory: true};
        Registration registration = 4; // SAPIENT Registration message
        RegistrationAck registration_ack = 5; // SAPIENT Registration Acknowledgement message
        StatusReport status_report = 6; // SAPIENT Status Report message
        DetectionReport detection_report = 7; // SAPIENT Detection Report message
        Task task = 8; // SAPIENT Task message
        TaskAck task_ack = 9; // SAPIENT Task Acknowledgement message
        Alert alert = 10; // SAPIENT Alert message
        AlertAck alert_ack = 11; // SAPIENT Alert Acknowledgment message
        Error error = 12; // SAPIENT Error message


	}
	optional string additional_information = 13; // space for additional information

}
// [END messages]
